util/kubeconfig/kubeconfig.go

```diff
@@ -51,6 +51,11 @@ func FromSecret(ctx context.Context, c client.Reader, cluster client.ObjectKey)
 
 // New creates a new Kubeconfig using the cluster name and specified endpoint.
 func New(clusterName, endpoint string, caCert *x509.Certificate, caKey *rsa.PrivateKey) (*api.Config, error) {
+	return NewPKCS8(clusterName, endpoint, caCert, caKey)
```

Comment:
Same, detecting a pattern here 😸

Reply:
+1 I think we can get away with updating `New` since it shouldn't break existing uses even if we are changing the signature.

Reply:
That was also a breaking change that go-apidiff job complained about. I am okay changing it, but then how should we handle the failing go-apidiff job ?